tests/system/test_cluster/test_agent_files_deletion/test_agent_files_deletion.py

```diff
@@ -33,13 +35,30 @@ def register_agent():
     yield
     host_manager.get_host(agent_host).ansible('command', f'service wazuh-agent restart', check=False)
 
+    # Wait until the agent is reconnected
+    timeout = time() + time_to_agent_reconnect
+    command = f'{WAZUH_PATH}/bin/cluster_control -a | grep active | wc -l'
+    while True:
+        if int(host_manager.run_shell('wazuh-worker2',
+                                      command)) == 4 and int(host_manager.run_shell('wazuh-master',
+                                                                                    command)) == 4 or time() > timeout:
+            break
+    sleep(time_to_sync)
+
 
 def test_agent_files_deletion(register_agent):
     """Check that when an agent is deleted, all its related files in managers are also removed."""
+    # Clean ossec.log and cluster.log
+    for hosts in managers_hosts:
+        host_manager.clear_file(host=hosts, file_path=os.path.join(WAZUH_LOGS_PATH, 'ossec.log'))
+        host_manager.clear_file(host=hosts, file_path=os.path.join(WAZUH_LOGS_PATH, 'cluster.log'))
+        host_manager.control_service(host=hosts, service='wazuh', state="restarted")
```

Comment:
If you want this to be run before the test starts, you can move it to the fixture `register_agent()` (you could even rename it to a more generic one), before line 35. Everything before `yield` in the fixture will be executed before starting the test.

In any case, this works here too, it's just a suggestion.